modules/API/external/external_api.js

```diff
@@ -720,6 +726,15 @@ export default class JitsiMeetExternalAPI extends EventEmitter {
         return avatarURL;
     }
 
+    /**
+     * Returns the absolute path of the default avatar.
+     *
+     * @returns {string}
+     */
+    getDefaultAvatarURL() {
+        return new URL('images/avatar.png', getBaseURL(this._frame.contentWindow)).href;
```

Comment:
Do we need this function? What is the value of it for the iframe API consumers?

If currently we are showing the default avatar in the iframe, can't we then just send this from the iframe?

Reply:
I'm not sure what you mean. This function was the main point of the discussion that we had earlier. This seems to be needed because we can't use relative paths to load an image (that the iFrame uses) and we can't detect the baseURL in the always on top window either, because that gets generated on top of "about:blank" by JS

Reply:
An alternative, which I'm not saying is better but is in my mind, is intercepting at the API.js level for avatar url changes and shoving in the default URL there. Or intercept even in the external_api middleware level to emit avatarUrl changes but with the default avatar url. I can say this so easily because I'm not doing the work, so I leave it to your judgement to tell me to shut up.

Reply:
@zbettenbuk What I think we agreed on is that this function is not very useful for any consumer of the API. And when the consumer calls `getAvatarURL` will expect to get an URL of an avatar that meet is showing in the iframe ( including the default one ) instead of having to implement a logic for the default avatar. Here I'm not concerned so much about the AoT implementation but for the iframe API.

@virtuacoplenny That was what I was thinking when we discussed with @zbettenbuk . I guess we'll also have to filter the use case where we are showing just letters for avatar and return undefined in this case.

Reply:
Thanks for the feedback guys! So yeah, I implemented the following:

- When we need to render an avatar, the avatar URL will be returned
- When initials need to be rendered, undefined is returned
- When default avatar needs to be rendered, the full URL of the default avatar is returned